charmcraft.yaml

```
@@ -12,3 +12,11 @@ parts:
   charm:
     build-packages:
       - git
+  promql-transform:
+    plugin: dump
+    source: .
+    build-packages:
+      - curl
+    override-pull: |
+      curl -L -O https://github.com/canonical/promql-transform/releases/download/rev6/promql-transform-${SNAP_ARCH}
+      chmod +x promql-transform-*
```

This is awesome. Just wondering:
- No `CRAFT_ARCH` yet? How is `SNAP_ARCH` set? Is it taken from "build-on"? And conveniently the string matches the release artifact?
- What made you change your mind about hardcoding the promql version?

Another potential option is:

```suggestion
  charm:
    build-packages:
      - git
      - curl
    override-build: |
      craftctl default
      curl -L -O https://github.com/canonical/promql-transform/releases/download/rev6/promql-transform-${SNAP_ARCH} -o $CRAFT_PART_INSTALL/promql-transform-${SNAP_ARCH}
      chmod +x $CRAFT_PART_INSTALL/promql-transform-*
```

---

Probably more of a case of messy documentatation. The bottom on [this page](https://craft-parts.readthedocs.io/en/latest/examples.html) wandered to the Snapcraft parts documentation, but `CRAFT_TARGET_ARCH` would work, too.

Mashing both of the parts into the same "part" (under `charm`) rather than keeping `promql-transform` discrete seems confusing to me.

Strictly speaking, we are still hardcoding to a specific release, just the most recent one.